pkg/datastore/kubernetes/kubernetes.go

```diff
@@ -31,8 +31,8 @@ type k8s struct {
 }
 
 type k8sSpecification struct {
```

Comment:
I think this should just be Specification - golint doesn't know that k8s is shortcut for kubernetes :)

Reply:
I fixed the two stuttering issues you mention in #42 ;-).

Reply:
> I think this should just be Specification - golint doesn't know that k8s is shortcut for kubernetes :)

That would export it, which I’m not sure is appropriate. The same exporting question goes for `k8s` too in the same class, which makes `golint` complain because it’s used as the return type for an exported function...

Reply:
yeah it doesn't need to be exported so we could rename to 'specification'. 

RE: k8s, NewK8sDatastore cannot return a *datastore.Datastore. It could return datastore.Datastore but then the method receivers would have to be values and not pointers. This seems like a strange aspect to Go - in Java we're used to a factory method being able to return an interface and hide the implementation class. You can do that in Go but it's not straightforward.  To eliminate that warning we could rename 'k8s' to 'Datastore' - that would export the type but none of the fields are exported.

Reply:
> yeah it doesn't need to be exported so we could rename to 'specification'.
> 
> RE: k8s, NewK8sDatastore cannot return a *datastore.Datastore. It could return datastore.Datastore but then the method receivers would have to be values and not pointers. This seems like a strange aspect to Go - in Java we're used to a factory method being able to return an interface and hide the implementation class. You can do that in Go but it's not straightforward. To eliminate that warning we could rename 'k8s' to 'Datastore' - that would export the type but none of the fields are exported.

Yup, the latter is exactly what I did in #49. I’ll update it to unexport `Specification`.